Dockerfile

```diff
@@ -0,0 +1,64 @@
+# First build step: check hash and unpack archive
+#
+# The hash ensures that args change even when the file name is equal, otherwise
+# docker would cache an image when GENERIC_BUILD_ARCHIVE does not change even
+# though the file it points to has changed.
+FROM adoptopenjdk:11-jre-hotspot as intermediate
+ARG GENERIC_BUILD_ARCHIVE
+ARG GENERIC_BUILD_SHA256
+
+COPY ${GENERIC_BUILD_ARCHIVE} /tmp/
+
+RUN export TMPDIR=$(mktemp -d) \
+    && mkdir -p /opt/rpki-validator-3 \
+    && if [ -z "$GENERIC_BUILD_SHA256" ]; then echo "Supply GENERIC_BUILD_SHA256"; exit 2; fi; \
+       echo -n "${GENERIC_BUILD_SHA256} /tmp/$(basename ${GENERIC_BUILD_ARCHIVE})" | sha256sum --check \ 
+    && tar -zxf /tmp/$(basename $GENERIC_BUILD_ARCHIVE) -C ${TMPDIR} \
+    # Move files from the dir in the archive (like rpki-validator-3.1-2020.01.13.09.31.26) to target folder:
+    && mv ${TMPDIR}/*/* /opt/rpki-validator-3
+
+# Second build step: Move files into place
+FROM adoptopenjdk:11-jre-hotspot
+# Keep the file name and sha256 in the metadata
+ARG GENERIC_BUILD_ARCHIVE
+ARG GENERIC_BUILD_SHA256
+LABEL validation.archive.file="$(basename ${GENERIC_BUILD_ARCHIVE})"
+LABEL validation.archive.sha256="${GENERIC_BUILD_SHA256}"
+
+# Webserver on 8080
+EXPOSE 8080
+
+# JVM memory settings
+ENV JVM_XMS=""
+ENV JVM_XMX=""
+# Used by `rpki-validator-3.sh`
+ENV CONFIG_DIR="/config"
+
+COPY --from=intermediate /opt/rpki-validator-3 /opt/rpki-validator-3
+WORKDIR /opt/rpki-validator-3
+
+RUN apt-get update && apt-get install --no-install-recommends --yes rsync \
+    # Clean apt cache
+    && rm -rf /var/lib/apt/lists/* \
```

Comment:
Wondering about this single RUN with multiple &&, and it seems the answer to which one we should follow changes over time :D 

https://stackoverflow.com/questions/39223249/multiple-run-vs-single-chained-run-in-dockerfile-what-is-better

Reply:
I'm using multistage builds. Otherwise the archive would be added to the disk (and add a layer) and later be unpacked. However, the file that you `ADD`/`COPY` never can be removed.

In the second step I need to adjust some parts of the config file. It is not an expensive layer that it would add but the layer is also not reusable if the archive changes. That is why I went to multiline-style (I'm used to it).